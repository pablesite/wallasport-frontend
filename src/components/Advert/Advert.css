   .card {
     background-color:#eeeeee;
   }

   .type {
      background-color:#ff5722;
    }

  .media {
     height: 20vh; 
  }

  .limit-height-12vh {
     height: 12vh;
  }

  .limit-height-6vh {
   height: 6vh;
}

.price-text{
   font-size: larger;
   margin-top: .5rem;
}