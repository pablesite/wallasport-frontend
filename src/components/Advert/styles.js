import { theme } from '../styles';

// const styledBy = (property, mapping) => props => mapping[props[property]];

export const styles = { 
  
  advertRoot: {
    margin: 5,
    width: 225,
    height: 330,
  },

  advertRootDetail: {
    margin: 5,
    width: 550,
  },

  advertMedia: {
    height: 0,
    paddingTop: '56.25%', // 16:9
    '&:hover': {
      cursor: 'pointer',
    }
  },

  advertReserved: {
    background: 'rgba(142, 231, 201, 0.35)',
  },

  advertSold: {
    background: 'rgba(142, 142, 142, 0.45)',
  },

  advertReservedText: {
    position: 'absolute', 
    transform: 'rotate(-45deg)',
    fontSize: 'xxx-large',
    top: '35vh',
    right: '44vw',
    zIndex: 1,
    color: theme.palette.primary.main,
  },

  advertSoldText: {
    position: 'absolute', 
    transform: 'rotate(-45deg)',
    fontSize: 'xxx-large',
    top: '35vh',
    right: '44vw',
    zIndex: 1,
  },

  advertUserOwner: {
    color: theme.palette.primary.main,
    display: 'inline',
    paddingLeft: '1em',

  },
  advertUserOwnerDetail: {
    color: theme.palette.primary.main,
    paddingTop: '1em',
    paddingLeft: '1em',
    fontSize: 'x-large',
  },

  advertAvatar: {
    width: 70,
    height: 70,
    border: '2px solid #fff',
    margin: '-43px 22px 0 auto',
    '& > img': {
      margin: 0,
    },
  },

  advertAvatarDetail: {
    width: 100,
    height: 100,
    border: '4px solid #fff',
    margin: '-86px 44px 0 auto',
    '& > img': {
      margin: 0,
    },
  },

  advertCardHeaderTitle: {
    color: 'inherit',
    fontSize: 'medium',
  },


  advertCardHeaderTitleDetail: {
    color: 'inherit',
    fontSize: 'x-large',
  },

  advertCardHeaderRoot: {
    paddingTop: 0,
    paddingBottom: 0,
  },

  advertCardHeaderRootDetail: {
    margin: 0,
  },

  advertAvatarSale: {
    backgroundColor: theme.palette.primary.main,
    width: 50,
    height: 50,
    fontSize: 'small'
  },

  advertAvatarSaleDetail: {
    backgroundColor: theme.palette.primary.main,
    width: 70,
    height: 70,
  },

  advertAvatarBuy: {
    backgroundColor: theme.palette.secondary.main,
    width: 50,
    height: 50,
    fontSize: 'small'
  },

  advertAvatarBuyDetail: {
    backgroundColor: theme.palette.secondary.main,
    width: 70,
    height: 70,
  },

  advertButton: {
    borderRadius: '20px 20px 20px 20px',
    textTransform: 'none',
  },

  advertDescription: {
    fontSize: 'small',
    maxHeight: 78,
    overflow: 'hidden',
  },

  advertDescriptionDetail: {
    fontSize: 'medium',
    minHeight: 100,
  },

  advertTags: {
    // nothin for the moment
  },

  advertTagsDetail: {
    // nothin for the moment
  },

  advertSocialNetwork: {
    marginRight:15,
  },


}