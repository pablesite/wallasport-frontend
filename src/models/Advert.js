export default class Advert {
  description;
  name;
  photo;
  price;
  tags;  
  type;
  _id
  

  constructor(value) {
    this.description = value.description;
    this.name = value.name;
    this.photo = value.photo;
    this.price = value.price;
    this.tags = value.tags;
    this.type = value.type;
    this._id = value._id;

  }

}

