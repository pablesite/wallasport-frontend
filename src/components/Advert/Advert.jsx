import React, { Component } from "react";
import { withRouter } from "react-router-dom";

import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import Grid from '@material-ui/core/Grid';

import "./Advert.css"

class Advert extends Component {
  goToDetail = () => {
    this.props.history.push(`/detail/${this.props.advert._id}`);
  };

  render() {
    const { advert } = this.props;
    return (
      <React.Fragment>
        <Grid
          onClick={this.goToDetail}
          item xs={10}
          sm={4}>

          <Card className="card">
            <div className="card">
              <CardHeader
                className="limit-height-12vh"
                avatar={
                  <Avatar aria-label="recipe"  className='avatar'>
                      {advert.type}
                  </Avatar>
                  
                }
                action={
                  <IconButton aria-label="settings">
                    <MoreVertIcon />
                  </IconButton>
                }

                title={advert.name}
                subheader={'Tags: ' + advert.tags.map((tags, i) => (
                  i === (advert.tags.length - 1) ? ` ${tags}.` : ` ${tags}`
                ))}

              />
              <CardMedia
                className="media"
                image={advert.photo !== 'noPhoto' ? `http://localhost:3001${advert.photo}` : `http://localhost:3000/noHayImagen.gif`}
                title={advert.name}
              />
              <CardContent>
                <Typography variant="body2" color="textSecondary" component="p" className="limit-height-6vh">
                  {advert.description}
                </Typography>
                
                <div className="price-text">
                    Precio: {advert.price} â‚¬.
                </div>
               
              </CardContent>
            </div>
          </Card>
        </Grid >
      </React.Fragment>
    );
  }
}

export default withRouter(Advert);
