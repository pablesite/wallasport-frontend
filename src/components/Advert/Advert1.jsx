import React, { Component } from "react";
import { withRouter } from "react-router-dom";

import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import Grid from '@material-ui/core/Grid';

import "./Advert.css"

export default function Advert(props) {


  const goToDetail = () => {
    // props.history.push(`/detail/${props.advert._id}`);
  };


  const { advert } = props;


  return (
    <React.Fragment>
      <Grid
        onClick={goToDetail}
        item
        xs={10}
        sm={3}>

        <Card className="card">

          {/* <div className="card"> */}

          <CardHeader
            className="limit-height-12vh"

            avatar={
              <Avatar aria-label="recipe" className='avatar'>
                {advert.type}
              </Avatar>
            }

            action={
              <IconButton aria-label="settings">
                <MoreVertIcon />
              </IconButton>
            }

            title={advert.name}

            subheader={'Tags: ' + advert.tags.map((tags, i) => (
              i === (advert.tags.length - 1) ? ` ${tags}.` : ` ${tags}`
            ))}

          />

          <CardMedia
            className="media"
            image={advert.photo !== 'noPhoto' ? `http://localhost:3003/${advert.photo}` : `http://localhost:3003/noHayImagen.gif`}
            title={advert.name}
          />

          <CardContent>
            <Typography variant="body2" color="textSecondary" component="p" className="limit-height-6vh">
              {advert.description}
            </Typography>

            <div className="price-text">
              Precio: {advert.price} â‚¬.
                </div>
          </CardContent>


          {/* </div> */}
        </Card>
      </Grid >
    </React.Fragment>
  );
}


// export default withRouter(Advert);
