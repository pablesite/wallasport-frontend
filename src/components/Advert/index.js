// Listo
import { connect } from 'react-redux';

import Advert from './Advert';
 import { getOneAdvert } from '../../store/actions';


function mapStateToProps(state) {
  
  return {
    // isFetching: state.ui.isFetching, 
    detail: state.adverts.detail
  };
}

const mapDispatchToProps = {
//   login: login,
goToDetail: getOneAdvert,

};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Advert);
