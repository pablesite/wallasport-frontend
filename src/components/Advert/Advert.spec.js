import React from 'react';
import { shallow } from 'enzyme';
import Advert from './Advert'

describe('Advert', () => {

    it('should render Advert component', () => {
        const props = {
            key: 0,

            advert: {
                _id: "5db5cd0a0a16ec2c68e2d96d",
                description: "Soy el del apartamento. Con eso lo digo todo.",
                name: "BMW X6 M",
                photo: "/images/anuncios/x6.jpg",
                price: 40000,
                tags: [{
                    0: "lifestyle",
                    1: "motor",
                }],
                type: "sell"
            }

        }
        const wrapper = shallow(<Advert.WrappedComponent {...props} />);

        expect(wrapper).toMatchSnapshot();

    });

});
