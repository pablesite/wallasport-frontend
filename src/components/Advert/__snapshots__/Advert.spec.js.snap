// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advert should render Advert component 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    item={true}
    onClick={[Function]}
    sm={4}
    xs={10}
  >
    <WithStyles(ForwardRef(Card))
      className="card"
    >
      <div
        className="card"
      >
        <WithStyles(ForwardRef(CardHeader))
          action={
            <ForwardRef(WithStyles)
              aria-label="settings"
            >
              <Memo(MoreVertIcon) />
            </ForwardRef(WithStyles)>
          }
          avatar={
            <ForwardRef(WithStyles)
              aria-label="recipe"
              className="avatar"
            >
              sell
            </ForwardRef(WithStyles)>
          }
          className="limit-height-12vh"
          subheader="Tags:  [object Object]."
          title="BMW X6 M"
        />
        <WithStyles(ForwardRef(CardMedia))
          className="media"
          image="http://localhost:3001/images/anuncios/x6.jpg"
          title="BMW X6 M"
        />
        <WithStyles(ForwardRef(CardContent))>
          <WithStyles(ForwardRef(Typography))
            className="limit-height-6vh"
            color="textSecondary"
            component="p"
            variant="body2"
          >
            Soy el del apartamento. Con eso lo digo todo.
          </WithStyles(ForwardRef(Typography))>
          <div
            className="price-text"
          >
            Precio: 
            40000
             â‚¬.
          </div>
        </WithStyles(ForwardRef(CardContent))>
      </div>
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
